treeherder/webapp/api/performance_data.py

```diff
@@ -492,3 +495,35 @@ def list(self, request):
 
         serializer = self.get_serializer(self.queryset, many=True)
         return Response(data=serializer.data)
+
+
+class ValidityDashboardViewSet(viewsets.ViewSet):
+
+    def list(self, request):
+        query_params = ValidityDashboardParamsSerializer(data=request.query_params)
+        if not query_params.is_valid():
+            return Response(data=query_params.errors,
+                            status=HTTP_400_BAD_REQUEST)
+
+        framework_id = query_params.validated_data['framework']
+        query_set = (PerformanceSignature.objects
+                     .prefetch_related('performancealert')
+                     .filter(framework_id=framework_id)
+                     .values('suite', 'test')
+                     .annotate(repositories=GroupConcat('repository_id', distinct=True))
+                     .annotate(platforms=GroupConcat('platform_id', distinct=True))
+                     .annotate(total_alerts=Count('performancealert'))
+                     .order_by('suite', 'test'))
+        return Response(self.__aggregate_signature_data(query_set))
+
+    def __aggregate_signature_data(self, query_set):
```

Comment:
I think it makes more sense to do this in the Serializer. You can do it with `to_representation`.